packages/devextreme/js/__internal/scheduler/header/m_view_switcher.test.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  afterEach, beforeEach, describe, expect, it, jest,
+} from '@jest/globals';
+import messageLocalization from '@js/common/core/localization/message';
+
+import type { SchedulerHeader } from './m_header';
+import { getDropDownViewSwitcher, getTabViewSwitcher } from './m_view_switcher';
+
+jest.mock('@js/common/core/localization/message', () => ({
+  format: jest.fn(),
+}));
+
+const mockSchedulerHeader = {
+  option: jest.fn(),
+  _updateCurrentView: jest.fn(),
+  _addEvent: jest.fn(),
+} as unknown as SchedulerHeader;
+
+const mockMessageLocalization = messageLocalization as jest.Mocked<typeof messageLocalization>;
+
+describe('m_view_switcher localization', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('localization change', () => {
+    it('should update view names when locale changes from English to Russian', () => {
```

Comment:
In this file we have a lot of mocks, that we will need to support in the future. For example if we want to change viewswticher a lot, then this test will fail, because mocks will become relevant.

So I suggest to implement integration test instead. In the neighbouring PR, I have created an [integration test file](https://github.com/DevExpress/DevExtreme/pull/31039/files#diff-398803b0bbcb8004d6f9bf902562012827a5376260d1a800c121ce86958aac10R1) for view switcher. 

What do you think?